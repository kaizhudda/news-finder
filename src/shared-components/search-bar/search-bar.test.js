import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from './index';

const props = {
  value: 'Test Search Value',
  onChange: jest.fn()
};
let wrapped = shallow(<SearchBar {...props} />);

describe('SearchBar', () => {
  it('should render the Search Bar Component', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should render the search value', () => {
    expect(wrapped.find('#search-bar').getElement().props.value).toEqual(props.value);
  });

  it('should call the onChange function when input field changed', () => {
    wrapped.find('#search-bar').simulate('change', { target: { value: props.value } });
    expect(props.onChange).toHaveBeenCalled();
  });
});