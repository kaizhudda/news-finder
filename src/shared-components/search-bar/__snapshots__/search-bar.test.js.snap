// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render the Search Bar Component 1`] = `
<form
  className="search-container"
>
  <input
    id="search-bar"
    onChange={[MockFunction]}
    placeholder="Search stories by title"
    type="text"
    value="Test Search Value"
  />
</form>
`;
